[New] Restyle Search Result cards

dynamic11 opened the pull request:
#### Changes introduced
- move action button to a ribbon at top of card
- added location to result card
- improves search animations
- show number of skills not visible
- make tag bubbles darker
- add counter for total results found

#### Related issue(s)
no related issues

#### Screenshots (if applicable)
**animation:**
![searchLoadAnimation](https://user-images.githubusercontent.com/11470442/93823649-2fdf1400-fc30-11ea-8f65-8fd5cbdff07c.gif)

**new look:**
![image](https://user-images.githubusercontent.com/11470442/93823698-45543e00-fc30-11ea-8a92-04055936afb9.png)

#### Checklist
If the database has been modified:
- [ ] Update database diagram <!-- Updated diagram located in the backend, at `./src/docs/I-Talent database.xml`, with draw.io and updated the png image at `./src/docs/I-Talent database.png` -->
- [ ] Create new migration <!-- Ran `yarn migrate:create` in backend docker container -->

If API endpoints has been modified:
- [ ] Update backend documentation <!-- Updated corresponding swagger documentation in the routers -->
- [ ] Create/Update validators for the API endpoints <!-- Restrict and sanitize user input in the routers with express-validator.github.io -->
- [ ] The API endpoints are properly secured with keycloak 
<!-- Use the `keycloak.protect(roleName)` express middleware in the routes -->

<!-- 
Optional for now, since tests are not working correctly
- [ ] Create tests for your changes
- [ ] Make sure the tests are passing 
-->

If the UI has been modified:
- [ ] It is translated in both language (with no hard coded text) <!-- To sort the keys and remove unused keys in the translation files, run `yarn i18n:cleanup` -->
- [ ] Has been tested on IE
- [ ] The modifications are tab friendly
- [ ] It is accessible

BenJeau commented on services/frontend-v3/src/components/resultsCard/ResultsCard.jsx:
```diff
@@ -31,7 +31,10 @@ const ResultsCard = () => {
       );
 
-      setResults(result.data);
+      setTimeout(() => {
+        setResults(result.data);
+      }, 500);
+
```
Should this be in the code?

dynamic11 replied:
i left it for PR testing I will remove before merge

dynamic11 replied:
removed

BenJeau commented on services/frontend-v3/src/i18n/fr_CA.json:
```diff
@@ -318,6 +318,10 @@
   "search.results.cards.remove.connection": "Déconnecter",
+  "search.results.cards.location.not.found": "Emplacement inconnu",
+  "search.results.cards.branch.not.found": "Branche inconnue",
+  "search.results.cards.skills.not.found": "Aucune compétence correspondante trouvée",
```
Since the english version is plural
```suggestion
  "search.results.cards.skills.not.found": "Aucunes compétences correspondantes trouvées",
```

BenJeau reviewed (changes requested):
Works well locally, small comments.

1. The content looks too "airy" to me, I would remove some padding in the card
2. When no jobtitle/classification is specified, it seems empty, maybe just add a "-"
![image](https://user-images.githubusercontent.com/22248828/93825623-a0d3fb00-fc33-11ea-8efa-089c3e9608c6.png)
to this?
![image](https://user-images.githubusercontent.com/22248828/93825667-bc3f0600-fc33-11ea-9d95-278448d8e485.png)
3. Saw you changed what the search API returns, could you also change the associated swagger UI docs? Should be as simple as adding a few properties to this https://github.com/CDH-Studio/UpSkill/blob/development/services/backend/src/docs/definitions.yml#L170, which makes references here https://github.com/CDH-Studio/UpSkill/blob/development/services/backend/src/router/search/docs.yml#L45

dynamic11 commented:
> Works well locally, small comments.
> 
> 1. The content looks too "airy" to me, I would remove some padding in the card
> 2. When no jobtitle/classification is specified, it seems empty, maybe just add a "-"
>    ![image](https://user-images.githubusercontent.com/22248828/93825623-a0d3fb00-fc33-11ea-8efa-089c3e9608c6.png)
>    to this?
>    ![image](https://user-images.githubusercontent.com/22248828/93825667-bc3f0600-fc33-11ea-9d95-278448d8e485.png)
> 3. Saw you changed what the search API returns, could you also change the associated swagger UI docs? Should be as simple as adding a few properties to this https://github.com/CDH-Studio/UpSkill/blob/development/services/backend/src/docs/definitions.yml#L170, which makes references here https://github.com/CDH-Studio/UpSkill/blob/development/services/backend/src/router/search/docs.yml#L45

- added documentation (please check if i did it correctly)
- reduced padding
-and added "-" for subtitle

BenJeau commented:
Looks good, the docs works, you can easily see them here http://localhost:8080/api-docs/#/search/get_api_search_filters. Could you rename `totalNumbResultSkills` to either `totalNumResultSkills` or `totalResultSkills`(numb sounds like a weird abbreviation)?

dynamic11 commented:
> Looks good, the docs works, you can easily see them here http://localhost:8080/api-docs/#/search/get_api_search_filters. Could you rename `totalNumbResultSkills` to either `totalNumResultSkills` or `totalResultSkills`(numb sounds like a weird abbreviation)?
> 
> Hope you don't mind that I pushed to your branch, its just that the fuzzy search API docs was not matching the actual response and to explain the changes via text would have been hard, feel free to take at the commit and ask me questions

done!

BenJeau commented on services/frontend-v3/src/components/resultsCard/ResultsCardView.jsx:
```diff
@@ -133,7 +133,7 @@ const ResultsCardView = ({
 
     if (person.groupLevel) {
-      subtitle = subtitle + " (" + person.groupLevel.name + ")";
+      subtitle = subtitle.concat(" (", person.groupLevel.name, ")");
```
I think string interpolation is used more with back ticks in the code, maybe changing it to this?

```suggestion
      subtitle = `${subtitle} (${person.groupLevel.name})`;
```

dynamic11 replied:
on it! 

I will leave this link for future reference:
https://stackoverflow.com/questions/27565056/es6-template-literals-vs-concatenated-strings/27567436
